### PR TITLE
Updating pillar settings for Juniper upgrade

### PR DESCRIPTION
**Pull request by blarghmatey**
- Migrated next residential settings to residential pillar
- Updated version specifiers for QA and production

### CODE REVIEW

File: pillar/edx/mitx-qa.sls [hunk 1]
```diff
@@ -5,3 +5,19 @@ edx:
...
   ansible_vars:
     EDXAPP_CAS_SERVER_URL: 'https://auth.mitx.mit.edu/cas'
+  dependencies:
+    os_packages:
+      - git
+      - libmysqlclient-dev
+      - mariadb-client-10.0
+      - landscape-common
+      - libssl-dev
+      - python2.7
```
Comment (shaidar) [inline pillar/edx/mitx-qa.sls:15]: Why 2.7?

File: pillar/edx/ansible_vars/residential.sls [hunk 2]
```diff
@@ -81,48 +86,15 @@ edx:
     EDXAPP_YOUTUBE_API_KEY: __vault__::secret-residential/global/edxapp-youtube-api-key>data>value
     EDXAPP_SUPPORT_SITE_LINK: 'https://odl.zendesk.com/hc/en-us/requests/new'
     EDXAPP_CACHE_OPTIONS:
-      'server_max_value_length': {{ memcached_server_max_value_length }}
+      'server_max_value_lengthn': {{ memcached_server_max_value_length }}
```
Comment (shaidar) [inline pillar/edx/ansible_vars/residential.sls:89]: Guessing the `n` at the end of length isn't a typo, but wanted to check

File: pillar/edx/ansible_vars/residential.sls [hunk 3]
```diff
@@ -174,11 +150,32 @@ edx:
...
+        db: gitlog_{{ purpose_suffix }}
+        host: mongodb-master.service.consul
+        user: __vault__:cache:mongodb-{{ environment }}/creds/gitlog-{{ purpose }}>data>username
+        password: __vault__:cache:mongodb-{{ environment }}/creds/gitlog-{{ purpose }}>data>password
+        replicaset: "{{ MONGODB_REPLICASET }}"
+        readPreference: "nearest"
+      SOCIAL_AUTH_SAML_SP_PRIVATE_KEY: __vault__::secret-residential/{{ environment }}/{{ purpose }}/saml-sp-cert>data>key
+      SOCIAL_AUTH_SAML_SP_PUBLIC_CERT: __vault__::secret-residential/{{ environment }}/{{ purpose }}/saml-sp-cert>data>value
+      {% if environment == 'mitx-qa' %}
+      PROCTORING_BACKENDS:
```
[the remark this answers is not in the document]
Comment (blarghmatey) [reply pillar/edx/ansible_vars/residential.sls:172]: Not really, just wasn't sure if it was useful to keep for reference. Happy to delete it.

File: pillar/edx/ansible_vars/residential.sls (hunk 2 above)
Comment (blarghmatey) [reply pillar/edx/ansible_vars/residential.sls:89]: Nope, that's a typo...

File: pillar/edx/mitx-qa.sls (hunk 1 above)
Comment (blarghmatey) [reply pillar/edx/mitx-qa.sls:15]: Just cargo culting

File: pillar/edx/ansible_vars/residential.sls (hunk 3 above)
[the remark this answers is not in the document]
Comment (shaidar) [reply pillar/edx/ansible_vars/residential.sls:172]: Yeah might as well. I can also come back and view the old config if I need to

**Comment by shaidar** [comment]
👍

**Pull request merged**